### PR TITLE
RC_Channel:make mount control metadata match param names

### PR DESCRIPTION
**Pull request by Hwurzburg**

### CODE REVIEW

**Comment by amilcarlucas** [comment]
Are the libcanard changes meant to be? Or an accident?

**Comment by Hwurzburg** [comment]
accident...will fix

**Comment by Hwurzburg** [comment]
> Looks ok. Should we rename "Retract Mount" to "Mount1 Retract"? (or Retract Mount1)
its not Mount1....its the primary mount, could be Mount2....I think we should add a switch for the second mount, and force them to only operate on a specific instance instead of whichever backend is created first....would remove ambiguity of primary, secondary, and eventually tertiary...
I added changes for the mount output servos...

**Comment by rmackay9** [comment]
@Hwurzburg,

yes, I agree we should change the unnumbered mount switches to always act on the first mount.

**Review by rmackay9 — APPROVED** [review]

**Review by rmackay9 — CHANGES_REQUESTED** [review]
@Hwurzburg,

Looking pretty good but a couple of small adminy type things:

1. split the commit into two commits.  One that affects RC_Channel and the other that affect SRV_Channel
2. prefix each commit with either "RC_Channel: " or "SRV_Channel: "

**Comment by Hwurzburg** [comment]
yep...forgot...done

File: libraries/RC_Channel/RC_Channel.cpp [hunk 1]
```diff
@@ -1197,21 +1197,21 @@ bool RC_Channel::do_aux_function(const aux_func_t ch_option, const AuxSwitchPos
...
             case AuxSwitchPos::HIGH:
-                mount->set_mode(MAV_MOUNT_MODE_RETRACT);
+                mount->set_mode(1,MAV_MOUNT_MODE_RETRACT);
                 break;
             case AuxSwitchPos::MIDDLE:
                 // nothing
                 break;
             case AuxSwitchPos::LOW:
-                mount->set_mode_to_default();
+                mount->set_mode_to_default(1);
```
Comment (tridge) [inline libraries/RC_Channel/RC_Channel.cpp:1214]: should be zero I think

File: libraries/RC_Channel/RC_Channel.cpp [hunk 2]
```diff
@@ -1197,21 +1197,21 @@ bool RC_Channel::do_aux_function(const aux_func_t ch_option, const AuxSwitchPos
...
+    case AUX_FUNC::RETRACT_MOUNT1: {
 #if HAL_MOUNT_ENABLED
         AP_Mount *mount = AP::mount();
         if (mount == nullptr) {
             break;
         }
         switch (ch_flag) {
             case AuxSwitchPos::HIGH:
-                mount->set_mode(MAV_MOUNT_MODE_RETRACT);
+                mount->set_mode(1,MAV_MOUNT_MODE_RETRACT);
```
Comment (tridge) [inline libraries/RC_Channel/RC_Channel.cpp:1208]: should be zero

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/RC_Channel/RC_Channel.cpp (hunk 1 above)
Comment (tridge) [inline libraries/RC_Channel/RC_Channel.cpp:1214]: should be zero

Comment (Hwurzburg) [reply libraries/RC_Channel/RC_Channel.cpp]: fixed

File: libraries/RC_Channel/RC_Channel.cpp (hunk 2 above)
Comment (Hwurzburg) [reply libraries/RC_Channel/RC_Channel.cpp]: fixed

File: libraries/RC_Channel/RC_Channel.cpp (hunk 1 above)
Comment (Hwurzburg) [reply libraries/RC_Channel/RC_Channel.cpp]: fixed

**Pull request merged**